MSVC Build broke in master on april 8th

I don't have a minimal repro at this time, sorry. 


Here are the build logs
```
..\third_party\range-v3\include\meta/meta.hpp(141): error C2794: 'type': is not a member of any direct or indirect base class of 'ranges::v3::adaptor_cursor<BaseIter,Adapt>::basic_adaptor_mixin_'
..\third_party\range-v3\include\range/v3/view_adaptor.hpp(266): note: see reference to alias template instantiation 'meta::_t<ranges::v3::adaptor_cursor<BaseIter,Adapt>::basic_adaptor_mixin_>' being compiled
..\third_party\range-v3\include\range/v3/view_adaptor.hpp(429): note: see reference to class template instantiation 'ranges::v3::adaptor_cursor<BaseIter,Adapt>' being compiled
```

--- a/include/range/v3/view_adaptor.hpp
+++ b/include/range/v3/view_adaptor.hpp
@@ -259,11 +259,7 @@ namespace ranges
             template<typename Adapt_>
             static meta::id<typename Adapt_::template mixin<basic_adaptor_mixin>> basic_adaptor_mixin_2_(int);
 
-            struct basic_adaptor_mixin_
-              : decltype(basic_adaptor_mixin_2_<Adapt>(42))
-            {};
-
-            using mixin = meta::_t<basic_adaptor_mixin_>;
+            using mixin = meta::_t<decltype(basic_adaptor_mixin_2_<Adapt>(42))>;
 
             using base_t::first;
             using base_t::second;

